constellations/gemini-admin/src/ask_for_help/afhEdit.js

```diff
@@ -0,0 +1,234 @@
+                .catch((err) => failCallback(err));
+        }
+    }
+
+    onClickCancel() {
+        window.location.hash = "afh";
+    }
+
+    onClickDelete() {
+        this.setState({ showDeleteModal: true });
+    }
+
+    onConfirmDelete() {
+        const afhId = this.state.id;
+        API.delete("api/askforhelp/afh/" + afhId).then((res) => {
+            window.location.hash = "afh";
+        });
+    }
+
+    onSavedOk() {
+        this.onDismissModal();
+        window.location.hash = "afh";
+    }
+
+    onDismissModal() {
+        this.setState({
+            showDeleteModal: false,
+            showSaveModal: false,
+        });
+    }
+
+    onMomentChange(newMoment) {
+        this.setState({ inputPostedAt: newMoment, date: newMoment });
+    }
+
+    render() {
+        const title =
+            (this.state.isEdit ? "Edit" : "Add") + " Ask For Help Session";
+
+        const optLocations = this.state.locations.map((loc, index) => (
+            <option key={index}>{loc.locationId}</option>
+        ));
+
+        let deleteButton = <div></div>;
+        if (this.state.isEdit) {
+            deleteButton = (
+                <button className="btn-delete" onClick={this.onClickDelete}>
+                    Delete
+                </button>
+            );
+        }
+
+        let modalDiv;
+        let modalContent;
+        let showModal;
+        if (this.state.showDeleteModal) {
+            showModal = this.state.showDeleteModal;
+            modalContent = (
+                <YesNoModal
+                    text={"Are you sure you want to delete?"}
+                    onAccept={this.onConfirmDelete}
+                    onReject={this.onDismissModal}
+                />
+            );
+        }
+        if (this.state.showSaveModal) {
+            showModal = this.state.showSaveModal;
+            modalContent = (
+                <OkayModal
+                    text={"Ask for help information saved!"}
+                    onOkay={this.onSavedOk}
+                />
+            );
+        }
+        if (modalContent) {
+            modalDiv = (
+                <Modal
+                    content={modalContent}
+                    show={showModal}
+                    onDismiss={this.onDismissModal}
+                />
+            );
+        }
+
+        return (
+            <div id="view-afh-edit">
+                {modalDiv}
+                <h2>{title}</h2>
+
+                <AFHEditDateTime
+                    postedAt={this.state.date}
+                    onMomentChange={this.onMomentChange}
+                />
+
+                <h3>Time</h3>
+                <input
+                    value={this.state.timeString}
+                    onChange={(e) => this.handleChange(e, "timeString")}
+                />
+
+                <h3>Title</h3>
+                <input
+                    value={this.state.title}
+                    onChange={(e) => this.handleChange(e, "title")}
+                />
```

Comment:
Take a look at the utils/InputText component that @tonywu315 made! Use it to replace the label / input functionality that you have here and you can get the validation logic + summaries/description stuff all for free!